contrib/externs/chrome_extensions.js

```diff
@@ -10262,3 +10262,116 @@ chrome.downloads.DeterminingFilenameEvent = function() {};
  * https://developer.chrome.com/extensions/downloads#event-onDeterminingFilename
  */
 chrome.downloads.onDeterminingFilename;
+
+
+/**
+ * @const
+ * @see https://developer.chrome.com/extensions/declarativeContent
+ */
+chrome.declarativeContent = {};
+
+
+/**
+ * @enum {string}
+ * @see https://developer.chrome.com/extensions/declarativeContent#type-PageStateMatcherInstanceType
+ */
+chrome.declarativeContent.PageStateMatcherInstanceType = {
+  DECLARATIVE_CONTENT_PAGE_STATE_MATCHER: ''
+};
+
+
+/**
+ * @enum {string}
+ * @see https://developer.chrome.com/extensions/declarativeContent#type-ShowPageActionInstanceType
+ */
+chrome.declarativeContent.ShowPageActionInstanceType = {
+  DECLARATIVE_CONTENT_SHOW_PAGE_ACTION: ''
+};
+
+
+/**
+ * @enum {string}
+ * @see https://developer.chrome.com/extensions/declarativeContent#type-SetIconInstanceType
+ */
+chrome.declarativeContent.SetIconInstanceType = {
+  DECLARATIVE_CONTENT_SET_ICON: ''
+};
+
+
+/**
+ * @enum {string}
+ * @see https://developer.chrome.com/extensions/declarativeContent#type-RequestContentScriptInstanceType
+ */
+chrome.declarativeContent.RequestContentScriptInstanceType = {
+  DECLARATIVE_CONTENT_REQUEST_CONTENT_SCRIPT: ''
+};
+
+
+/**
+ * @typedef {{
```

Comment:
See section C of the "Best Practices" documentation comment at the top of this file.
All typedefs need to be declared with `{!{`
Thanks.